src/lib.rs

```diff
@@ -1372,13 +1372,13 @@ impl<A: Array> SmallVec<A> {
             }
             let mut ptr = ptr.as_ptr();
             let len = *len_ptr;
+            if index > len {
+                panic!("index exceeds length");
+            }
```

Comment:
This panic placement is load-bearing and probably should have a SAFETY comment:

```suggestion
            }
            // SAFETY: add is UB if index > len, but we panicked first
```